src/relay/qnn/util.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file tvm/relay/quantize_util.h
+ * \brief Utility methods needs for quantized ops that can be shared
+ */
+
+#ifndef TVM_RELAY_QUANTIZE_UTIL_H_
+#define TVM_RELAY_QUANTIZE_UTIL_H_
+
+#include <tvm/expr.h>
+#include <limits>
+#include <tvm/relay/expr.h>
+
+namespace tvm {
+namespace relay {
+
+inline bool IsInt8(const DataType& dtype) {
+  return dtype == Int(8);
+}
+
+inline bool IsUint8(const DataType& dtype) {
+  return dtype == UInt(8);
+}
+
+inline bool IsInt16(const DataType& dtype) {
+  return dtype == Int(16);
+}
+
+inline bool IsUint16(const DataType& dtype) {
+  return dtype == UInt(16);
+}
+
+inline bool IsInt32(const DataType& dtype) {
+  return dtype == Int(32);
+}
+
+inline bool IsUint32(const DataType& dtype) {
+  return dtype == UInt(32);
+}
+
+inline bool IsFloat32(const DataType& dtype) {
+  return dtype == Float(32);
+}
+
+inline bool IsQuantizedType(const DataType& dtype) {
+  return IsInt8(dtype) || IsUint8(dtype)
+      || IsInt16(dtype) || IsUint16(dtype);
+}
+
+enum class QuantizeOpType : uint8_t {
```

Comment:
unless there is a need(data compactness), you don't want to use uint8_t as storage type